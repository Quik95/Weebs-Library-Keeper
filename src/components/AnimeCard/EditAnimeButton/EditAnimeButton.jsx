import React, { memo } from 'react';
import PropTypes from 'prop-types';

// MUI
import Fab from '@material-ui/core/Fab';
import EditIcon from '@material-ui/icons/Edit';
import Tooltip from '@material-ui/core/Tooltip';

// styles
import useStyles from './EditAnimeButton.styles';

const EditAnimeButton = memo(function EditAnimeButton(props) {
  const classes = useStyles();

  return (
    <Tooltip
      title="Click to edit anime"
      aria-label="Edit anime"
      data-testid="EditAnimeButton"
    >
      <Fab
        size="small"
        color="secondary"
        aria-label="Edit anime"
        className={classes.editButton}
      >
        <EditIcon className={classes.editIcon} />
      </Fab>
    </Tooltip>
  );
});

EditAnimeButton.propTypes = {};

export default EditAnimeButton;
