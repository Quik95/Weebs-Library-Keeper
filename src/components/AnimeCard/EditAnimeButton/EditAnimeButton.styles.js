import { makeStyles } from '@material-ui/core/styles';

export default makeStyles(theme => ({
  editButton: {
    position: 'absolute',
    bottom: '-20px',
    right: '-20px',
  },
  editIcon: {
    color: 'white',
  },
}));
