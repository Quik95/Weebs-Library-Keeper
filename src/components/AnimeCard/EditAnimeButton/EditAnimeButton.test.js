import EditAnimeButton from './EditAnimeButton';

describe('<EditAnimeButton/>', () => {
  it('Renders correctly', () => {
    const wrapper = shallow(<EditAnimeButton />);
    expect(wrapper).toMatchSnapshot();
  });
});
