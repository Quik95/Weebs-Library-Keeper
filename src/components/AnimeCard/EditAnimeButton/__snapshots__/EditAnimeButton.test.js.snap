// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditAnimeButton/> Renders correctly 1`] = `
<WithStyles(ForwardRef(Tooltip))
  aria-label="Edit anime"
  data-testid="EditAnimeButton"
  title="Click to edit anime"
>
  <WithStyles(ForwardRef(Fab))
    aria-label="Edit anime"
    className="makeStyles-editButton-1"
    color="secondary"
    size="small"
  >
    <EditIcon
      className="makeStyles-editIcon-2"
    />
  </WithStyles(ForwardRef(Fab))>
</WithStyles(ForwardRef(Tooltip))>
`;
